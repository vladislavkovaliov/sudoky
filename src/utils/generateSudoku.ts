import { GRID_SIZE } from "../constant";

export function generateSudoku() {
  const sudoku = generateEmptyGrid();

  if (import.meta.env.VITE_DEBUG_MODE === "true") {
    console.table(sudoku);
  }
}

export function generateEmptyGrid() {
  return new Array(GRID_SIZE)
    .fill(null)
    .map(() => new Array(GRID_SIZE).fill(null));
}
