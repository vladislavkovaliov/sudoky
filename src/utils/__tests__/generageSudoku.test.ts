import { describe, test, expect } from "vitest";
import { generateSudoku } from "../generateSudoku.ts";

describe("[generateSudoku.ts]", () => {
  test("should generate empty sudoku grid", () => {
    const result = generateSudoku();

    expect(result).toMatchSnapshot();
  });
});
