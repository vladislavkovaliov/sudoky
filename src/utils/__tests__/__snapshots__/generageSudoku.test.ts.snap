// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[generateSudoku.ts] > should generate empty sudoku grid 1`] = `undefined`;
